### PR TITLE
dataclients/kubernetes: add TrafficSegment-based backend traffic calculator

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
* adds backend traffic split algorithm implementation based on TrafficSegment predicate
* adds BackendTrafficAlgorithm k8s dataclient option (for now only available in tests)
* adds basic test suite for the implementation

This is part 1 of N in series of changes.

For #2268

### CODE REVIEW

File: dataclients/kubernetes/kube.go [hunk 1]
```diff
@@ -215,6 +215,9 @@ type Options struct {
 	// ForceKubernetesService overrides the default Skipper functionality to route traffic using
 	// Kubernetes Endpoint, instead using Kubernetes Services.
 	ForceKubernetesService bool
+
+	// BackendTrafficAlgorithm specifies the algorithm to calculate the backend traffic.
+	BackendTrafficAlgorithm string
```
Comment (AlexanderYastrebov) [inline dataclients/kubernetes/kube.go:220]: The plan is to make this configurable via flag.

Comment (RomanZavodskikh) [reply dataclients/kubernetes/kube.go:220]: What are potential values for this flag? Random, Round Robin, etc?

Comment (AlexanderYastrebov) [reply dataclients/kubernetes/kube.go:220]: I named the new algorithm `"traffic-segment-predicate"` and the current default would be `"traffic-predicate"`

**Comment by RomanZavodskikh** [comment]
:

**Comment by RomanZavodskikh** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**